### ISSUE TITLE
deadzone math

### ISSUE THREAD

**Issue by briaguya0** [Kenix3/libultraship#351]
i would like to allow users to set a deadzone with a unit that means something. the current deadzone is just a number with no context. i think it would be ideal to have this number be a percentage, with 100% meaning no inputs from the stick are counted because 100% of the stick range is included in the deadzone.

if SDL always reported a value that fit within a circle this wouldn't be a problem. `sqrt(x^2 + x^2)` would be at max `MAX_AXIS_RANGE` and we could just have the user set a percentage of `MAX_AXIS_RANGE`, but we can't count on that being the case.

in my testing on an xbox controller (with a circular gate on the analog stick), i've been able to get normalized values (`MAX_AXIS_RANGE/MAX_SDL_RANGE`), that give a radius of closer to `90` (as opposed to `85`) (~64 and ~63 for x and y respectively)

for https://github.com/Kenix3/libultraship/pull/350 i'm going to go ahead and just do a percentage of `MAX_AXIS_RANGE`, as that still feels better than what we have now

![image](https://github.com/Kenix3/libultraship/assets/70942617/5f275495-eeae-4128-befd-47771003e1b6)

# tl;dr

try setting the deadzone in the current controller config window somewhere between 85 and 95 and look at the input preview, you'll see the oddities
